lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -29,32 +29,34 @@ def self.run(params)
         auth_token = fetch_auth_token(params[:service_credentials_file], params[:firebase_cli_token])
         fad_api_client = Client::FirebaseAppDistributionApiClient.new(auth_token, params[:debug])
 
-        # If binary is an AAB get FULL view of app which includes the aab_state
-        app_view = binary_type == :AAB ? 'FULL' : 'BASIC'
-        app = fad_api_client.get_app(app_id, app_view)
-        validate_app!(app, binary_type)
-        release_id = fad_api_client.upload(app.project_number, app_id, binary_path, platform.to_s)
+        # If binary is an AAB, get the AAB info for this app, which includes the integration state and certificate data
+        if binary_type == :AAB
+          aab_info = fad_api_client.get_aab_info(app_id)
+          validate_aab_setup!(aab_info)
+        end
+
+        release_id = fad_api_client.upload(app_id, binary_path, platform.to_s)
 
-        if binary_type == :AAB && app.aab_certificate.empty?
-          updated_app = fad_api_client.get_app(app_id)
-          unless updated_app.aab_certificate.empty?
+        if binary_type == :AAB && aab_info && !aab_info.certs_provided?
+          updated_aab_info = fad_api_client.get_aab_info(app_id)
+          if updated_aab_info.certs_provided?
             UI.message("After you upload an AAB for the first time, App Distribution " \
               "generates a new test certificate. All AAB uploads are re-signed with this test " \
               "certificate. Use the certificate fingerprints below to register your app " \
               "signing key with API providers, such as Google Sign-In and Google Maps.\n" \
-              "MD-1 certificate fingerprint: #{updated_app.aab_certificate.md5_certificate_hash}\n" \
-              "SHA-1 certificate fingerprint: #{updated_app.aab_certificate.sha1_certificate_hash}\n" \
-              "SHA-256 certificate fingerprint: #{updated_app.aab_certificate.sha256_certificate_hash}")
+              "MD-1 certificate fingerprint: #{updated_aab_info.md5_certificate_hash}\n" \
+              "SHA-1 certificate fingerprint: #{updated_aab_info.sha1_certificate_hash}\n" \
+              "SHA-256 certificate fingerprint: #{updated_aab_info.sha256_certificate_hash}")
           end
         end
 
-        fad_api_client.post_notes(app_id, release_id, release_notes(params))
+        fad_api_client.update_release_notes(app_id, release_id, release_notes(params))
 
         testers = get_value_from_value_or_file(params[:testers], params[:testers_file])
         groups = get_value_from_value_or_file(params[:groups], params[:groups_file])
         emails = string_to_array(testers)
-        group_ids = string_to_array(groups)
-        fad_api_client.enable_access(app_id, release_id, emails, group_ids)
+        group_aliases = string_to_array(groups)
+        fad_api_client.enable_access(app_id, release_id, emails, group_aliases)
```

Comment:
rename `enable_access` to `distribute` or `distribute_release`?

Reply:
Good idea. I like `distribute` since it's parallel to `upload`.